modules/imgcodecs/src/grfmt_tiff.cpp

```diff
@@ -817,6 +821,16 @@ bool TiffEncoder::writeLibTiff( const std::vector<Mat>& img_vec, const std::vect
             return false;
         }
 
+        if (((resUnit >= RESUNIT_NONE && resUnit <= RESUNIT_CENTIMETER) && !TIFFSetField(pTiffHandle, TIFFTAG_RESOLUTIONUNIT, resUnit))
```

Comment:
`RESUNIT_NONE` and `RESUNIT_CENTIMETER` are defined in the libtiff dev header. Caller of imwrite can use those constants as flags to `imwrite` to specify legal values of resolution units.